### PR TITLE
Mona - Fire

### PR DESCRIPTION
**Pull request by MonaRahmani**
# Task List
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe in your own words what the Model is doing in Rails | model linked to the database. it can hold the data of the application.
Describe in your own words what the Controller is doing in Rails | controller is the brain of the rails app. Receive HTTP requests, gather data from the model, pass data to the view to be rendered as HTML.
Describe in your own words what the View is doing in Rails | View Turns data into HTML, only display logic, and does not contain "business logic". 
Describe an edge-case controller test you wrote | Edge Case for the destroy Action: If the instance is not in the database, expect to see a response of 404 or :not_found
What is the purpose of using strong params? (i.e. the params method in the controller) | helps specify what data can be submitted by using require and permit, also helps DRY code.
How are Rails migrations related to Rails models? | they can edit the schema 
Describe one area of Rails that are still unclear on | rendering partials in rails.

### CODE REVIEW

File: app/controllers/tasks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    if task.nil?
+      head :not_found
+      return
+    else
+      task.destroy
+      redirect_to tasks_path
+    end
+  end
+
+  def mark_complete
```
Comment (beccaelenzil) [inline app/controllers/tasks_controller.rb:77]: Consider how making two separate methods, for instance `mark_complete` and `mark_incomplete` would make this action idempotent.

File: app/views/tasks/_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<%= form_with model: @task, class: 'create-task' do |f|%>
```
Comment (beccaelenzil) [inline app/views/tasks/_form.html.erb:1]: Good use of a partial view.

File: app/views/tasks/_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    <%= f.text_field :name %>
+  </div>
+
+  <div>
+    <%= f.label :description %>
+    <%= f.text_area :description %>
+  </div>
+
+  <div>
+    <%= f.label :completed_at %>
```
Comment (beccaelenzil) [inline app/views/tasks/_form.html.erb:14]: Consider automatically populating `completed_at` when a task is created. Additionally, this does not need to be in the `edit` view since `mark_complete` takes care of changing this.

File: app/controllers/tasks_controller.rb (hunk 1 above)
Comment (beccaelenzil) [reply app/controllers/tasks_controller.rb:77]: Currently the user doesn't have the ability to mark a task incomplete

**Review by beccaelenzil — COMMENTED** [review]
# Task List




## Major Learning Goals/Code Review





| Criteria                                           | yes/no, and optionally any details/lines of code to reference |
| -------------------------------------------------- | ------------------------------------------------------------- |
| At least 6 commits with meaningful commit messages | ✔️                                                             |
| Routes follow RESTful conventions                  | ✔️                                                             |
| Uses named routes (like `_path`)                   | ✔️                                                            |
| Creates Models and migrations                      | ✔️                                                            |
| Creates styled views                               | Not required                                                            |
| Handles errors like nonexistant tasks              | ✔️                                                            |
| Uses `form_with` to render forms in Rails          | ✔️                                                            |

## Functional Requirements/Manual Testing

| Functional Requirement                                                         | yes/no |
| ------------------------------------------------------------------------------ | ------ |
| Successfully handles index & show                                              | ✔️     |
| index & show tests pass                                                        | ✔️     |
| Successfully handles: New, Create                                              | ✔️     |
| New, Create tests pass                                                         | ✔️     |
| Successfully handles: Edit, Update                                             | ✔️     |
| Edit, Update tests pass with valid & invalid task ids                          | ✔️     |
| Successfully handles: Destroy, Task Complete                                   | There is not ability to mark incomplete     |
| Tests for Destroy & Task Complete include tests for valid and invalid task ids | ✔️     |

## Overall Feedback

Great work on this project. You've created your first web app in rails, nice work! It is clear that the learning goals around using rails conventions, strong params, and partial views were met. I've left a few in line comments for you to review. Keep up the hard work!


| Overall Feedback                | Criteria                                                                                                                                                                                      | yes/no |
| ------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------ |
| Green (Meets/Exceeds Standards) | 5+ in Code Review && 6+ in Functional Requirements                                                                                                                                            |✔️
| Yellow (Approaches Standards)   | 3+ in Code Review && 5+ in Functional Requirements, or the instructor judges that this project needs special attention                                                                        |
| Red (Not at Standard)           | 0-2 in Code Review or 0-4 in Functional Reqs, or assignment is breaking/doesn’t run with less than 5 minutes of debugging, or the instructor judges that this project needs special attention |





## Code Style Bonus Awards



Was the code particularly impressive in code style for any of these reasons (or more...?)

| Quality              | Yes? |
| -------------------- | ---- |
| Perfect Indentation  | ✅    |
| Descriptive/Readable | ✅    |
| Concise              | ✅    |
| Logical/Organized    | ✅    |
